server/extensions/component-v2.js

```diff
@@ -0,0 +1,21 @@
+import nunjucks from 'nunjucks';
+
+export default class Component {
+  constructor(env) {
+    this.tags = ['componentV2'];
+    this.env = env;
+  }
+
+  parse(parser, nodes, /*lexer*/) {
+    const token = parser.nextToken();
+    const args = parser.parseSignature(null, true);
+    parser.advanceAfterBlockEnd(token.value);
+
+    return new nodes.CallExtension(this, 'run', args);
+  };
+
+  run(context, name, model, modifiers = [], data = {}) {
```

Comment:
As a note - this model cannot be a `Immutable.Record` due to nunjucks not handling getters / setters all that well. I'd like to dig into this at some stage to help give back etc.